ml/backend.go

```diff
@@ -144,6 +144,8 @@ type Tensor interface {
 	Conv2D(ctx Context, weight Tensor, s0, s1, p0, p1, d0, d1 int) Tensor
 
 	RoPE(ctx Context, positionIDs, ropeFactors Tensor, dim, ropeType uint32, base, scale float32) Tensor
+	// RoPEWithLen allows the caller to specify the rope default context length
+	RoPEWithLen(ctx Context, positionIDs, ropeFactors Tensor, ropeDim, ropeType, defaultContextLen uint32, ropeBase, ropeScale float32) Tensor
```

Comment:
Not a huge fan of this, but it focuses this PR on the actual model implementation.

We can improve the RoPE interface in a different change and come back to this (ex: #10078)